### ISSUE TITLE
Forward cookies get from HTTP call to Websocket calls

### ISSUE THREAD

**Issue by mburger81** [silkimen/cordova-plugin-advanced-http#6]
The plugin works good with session cookies, they are forwarded on every http or https call as expected.

But In our project we use also WebSockets, and we need to forward the cookies get from http/s call with your plugin to the WebSocket call.

If we use native http calls and native websockets calls this is working out of the box, I'm not sure if this should also work in your case, but I don't think so. Also I don't know how you are handling the cookie store and how we can extract the cookie.

So what do you think about it, can we do this with your plugin, should it work, or do you think you can adapt the plugin working for that?

**Comment by silkimen** [comment]
Ok, as for WebSockets: this plugin is not meant to support WebSockets, so you'll need to stick to the native WebSockets or find a plugin supporting WebSockets.

As for cookie handling: we are using the UMD version of module "tough-cookie" (see https://github.com/silkimen/umd-tough-cookie) for cookie handling, so basically we could build an interface for accessing the raw cookies. But this was not the purpose of this plugin.

Do you have an idea how this interface should look like? I could provide an interface exposing the cookie jar. So you could implement your own one. Is that what you are looking for?

**Comment by mburger81** [comment]
Yes I know this project is not intend to use for Websocket, but I have to do. 
We are already using a native websocket cordova plugin for Android. Perhapse I can merge the two projects and we can create a new one more complete plugin. What do you think. 

Not sure about about the cookies I'm not the expert on it. But I think we have to read the cookies for a url and also remove and add now cookies comming from a websocket. 
So perhapse a function getCookieStingForUrl should be a good point to start??
What do you think?

**Comment by silkimen** [comment]
If you want to create a new plugin containing WebSocket and HTTP functionality, you can fork this plugin and then use "setCookieFromString" and "getCookieString" [see here](https://github.com/silkimen/cordova-plugin-advanced-http/blob/8ac8b952466fc121d91ff4a51b9874d684a45373/www/cookie-handler.js#L38-L50). But you really should check the API documentation of [tough-cookie](https://github.com/salesforce/tough-cookie) first. 

As for this plugin, it makes no sense to expose those cookies.

**Comment by mburger81** [comment]
I had a look to other implementations for cordova, perhaps it could be better to use 
`android.webkit.CookieManager` for the cookie management. I know you have to do there two different implementations one for ios and one for android, but this should have less bugs and less code and much simpler to maintain.
And so all other plugins which are using default native cookie manager could use cookies, so it would be more trasversal. Could this be an idea?

**Comment by pc-wurm** [comment]
@mburger81 May I ask how do you make the requests so that the cookies are sent? I cannot get it simply work.

I make a post request to the backend server with the credentials and I can see in the backend logs that the session successfully opened. But the subsequent requests returns that the user is not logged in.

The same code works using angular http, but we need SSL-Pinning, so we must use native http.
I don't know what I'm missing.

**Comment by matodrobec** [comment]
@pc-wurm  I have same problem with session cookies. If I am using angular XHRBackend then it is working well.  Do you find way how to  handle session cookie? Cookie from server look like `'set-cookie': [ 'JSESSIONID=xxxxxxxxxxxx; Path=/api; Secure' ],``
